### DISCUSSION TITLE
Should package version match assembly version?

### DISCUSSION THREAD

**Discussion by SMayer135** [nhibernate/nhibernate-core#3587] [Q&A] [answered]
I've updated NH in my solution to 5.5.2 to address the recent vulnerabilities, but it appears the the assembly is still marked as 5.5.0 (for package 5.5.2). 5.4.9 is marked as 5.4.0. Since this vulnerability is only fixed in these specific patch versions, shouldn't the assembly be updated to indicate that?

**Comment by fredericDelaporte** [comment, marked as answer]
No, it should not. It allows the patched assembly to be used as a dropped-in replacement, without recompiling the consuming applications, especially with .Net Framework.

See [this StackOverflow answer](https://stackoverflow.com/a/3769491/1178314) and other answers to the question, for more on the subject.
